### PR TITLE
Api query encoding/decoding and prev cursor patch

### PR DESCRIPTION
**Pull request by kfangw**
### Github Issue
#819 #820 #823 

### Background
See issue #819 

### Solution
Encode the cursor with base64 before encoding to query string
Decode the cursor with base64 when parsing the query string

cursor for the `next` `prev` link needs to be different.

LevelDB iterator refactoring #823

### CODE REVIEW

**Comment by kfangw** [comment]
the integration_tests_sdk fail is not relevant to this PR.
I am working on the fail in another branch.

**Comment by kfangw** [comment]
@anarcher @spikeekips 
Could you review this PR please?

File: lib/node/runner/api/account.go [hunk 1]
```diff
@@ -221,10 +226,13 @@ func (api NetworkHandlerAPI) GetFrozenAccountsHandler(w http.ResponseWriter, r *
 			)
 
 			bo, hasNext, c := iterFunc()
-			cursor = c
 			if !hasNext {
 				break
 			}
+			cursor = c
+			if len(firstCursor) == 0 {
+				firstCursor = append(firstCursor, c...)
```
Comment (spikeekips) [inline lib/node/runner/api/account.go]: `firstCursor = c` is not enough?

**Comment by kfangw** [comment]
@anarcher @spikeekips
Could you confirm the latest commit? 4a5fceb
It is related to #823

**Pull request merged**